diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index b4fade01..8e3b9dc3 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -97,6 +97,7 @@ target_include_directories( jit PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/../include>
         $<INSTALL_INTERFACE:include> )
 target_include_directories(jit PUBLIC ${LLVM_INCLUDE_DIRS} )
+list(FILTER LLVM_AVAILABLE_LIBS INCLUDE REGEX "LLVM(MCJIT|.+(CodeGen|AsmParser))" )
 target_link_libraries( jit PUBLIC ${LLVM_AVAILABLE_LIBS} )
 target_link_libraries( jit PUBLIC QMP::qmp)
 
