diff --git a/include/targets/hip/block_reduction_kernel.h b/include/targets/hip/block_reduction_kernel.h
index e82d6f2df..d81c213dd 100644
--- a/include/targets/hip/block_reduction_kernel.h
+++ b/include/targets/hip/block_reduction_kernel.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <target_device.h>
+#include <constant_kernel_arg.h>
 #include <reduce_helper.h>
 
 namespace quda
diff --git a/include/targets/hip/kernel.h b/include/targets/hip/kernel.h
index 341bb888d..3862361e7 100644
--- a/include/targets/hip/kernel.h
+++ b/include/targets/hip/kernel.h
@@ -2,6 +2,7 @@
 #include <hip/hip_runtime.h>
 #include <kernel_helper.h>
 #include <target_device.h>
+#include <constant_kernel_arg.h>
 
 namespace quda
 {
diff --git a/include/targets/hip/reduction_kernel.h b/include/targets/hip/reduction_kernel.h
index 3bccf1458..5590df2ac 100644
--- a/include/targets/hip/reduction_kernel.h
+++ b/include/targets/hip/reduction_kernel.h
@@ -1,5 +1,6 @@
 #pragma once
 #include <target_device.h>
+#include <constant_kernel_arg.h>
 #include <reduce_helper.h>
 
 namespace quda
diff --git a/include/targets/hip/target_device.h b/include/targets/hip/target_device.h
index bcc430dbc..0ca7fb365 100644
--- a/include/targets/hip/target_device.h
+++ b/include/targets/hip/target_device.h
@@ -172,7 +172,7 @@ namespace quda
        the kernel arguments passed to a kernel on the target
        architecture.
     */
-    constexpr size_t max_kernel_arg_size() { return 4096; }
+    constexpr size_t max_kernel_arg_size() { return 0; }
 
     /**
        @brief Helper function that returns true if we are to pass the
